Apsitvarkom.DataAccess/PollutedLocationDatabaseRepository.cs

```diff
@@ -21,21 +21,25 @@ public PollutedLocationDatabaseRepository(IPollutedLocationContext context)
     /// <inheritdoc />
     public async Task<IEnumerable<PollutedLocation>> GetAllAsync()
     {
-        return await _context.PollutedLocations.ToListAsync();
+        return await _context.PollutedLocations
+            .Include(l => l.Events)
+            .ToListAsync();
     }
 
     /// <inheritdoc />
     public async Task<IEnumerable<PollutedLocation>> GetAllAsync(Coordinates inRelationTo)
     {
-        return from pollutedLocation in await _context.PollutedLocations.ToListAsync()
+        return from pollutedLocation in await _context.PollutedLocations.Include(l => l.Events).ToListAsync()
             orderby pollutedLocation.Location.Coordinates.DistanceTo(inRelationTo)
             select pollutedLocation;
     }
 
     /// <inheritdoc />
-    public Task<PollutedLocation?> GetByPropertyAsync(Expression<Func<PollutedLocation, bool>> propertyCondition)
+    public async Task<PollutedLocation?> GetByPropertyAsync(Expression<Func<PollutedLocation, bool>> propertyCondition)
     {
-        return _context.PollutedLocations.FirstOrDefaultAsync(propertyCondition);
+        return await _context.PollutedLocations
+            .Include(l => l.Events)
+            .FirstOrDefaultAsync(propertyCondition);
```

Comment:
`async` `await` is not needed here, it just adds overhead, see https://stackoverflow.com/a/38017167/13135665

As a rule of thumb, if you can directly return a `Task`, you should. Might be worth checking other methods also